src/main/java/io/openshift/boosters/FruitResource.java

```diff
@@ -82,8 +87,9 @@ public Response create(Fruit fruit) {
     @Produces("application/json")
     @Transactional
     public Response update(@PathParam("id") Integer id, Fruit fruit) {
-        if (fruit.getName() == null) {
-            return error(422, "Fruit Name was not set on request.");
+        
+        if (fruit == null || fruit.getName() == null || fruit.getName().trim().length() == 0) {
+            return error(400, "The name is required!");
```

Comment:
I think 422 is more appropriate.

Reply:
Nodejs and Vert.x use 400, but for the e2e tests it isn't important to have same status code for this use case so I will fix it